### PR TITLE
[UG] Reformat find notes

### PR DESCRIPTION
**Pull request by lingshanng**

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -440,79 +457,113 @@ This section guides you on how to find or filter students of your choice in TAB.
...
+Example:
+* `find t/unpaid t/New` will return all students with both of the tags `UNPAID` and `NEW`.
 
-* The search is case-insensitive.<br>
-  e.g. keyword `hans` will match `Hans`.
+##### Search by other student fields
+* You can specify one or more keywords for each parameter. e.g. `find n/Amad Ali`.
+* A field needs to contain all specified keywords to be matched.<br>
+  e.g. keywords `Amad Ali` will not match `Amad`, but it will match `Amad bin Ali`.
+* If you specify multiple of the same prefix, only the last prefix will be used for the search of that field.
```
Comment (Xiaoyunnn) [inline docs/UserGuide.md:480]:
```suggestion
* If you specify multiple keywords with the same prefix, only the last prefix will be used to search that field.
```
Not sure if this sounds better

File: docs/UserGuide.md [hunk 2]
```diff
@@ -542,121 +586,63 @@ A lesson can be categorised into 2 types:
...
-            <td><ul>
-              <li><code>hw/Test 2</code> is valid</li>
-              <li><code>hw/</code> clears <strong>all</strong> existing pieces of homework.</li>
-            <li><code>hw/ hw/Worksheet 1</code> removes the existing pieces of homework and adds <code>Worksheet 1</code> to the homework set.</li>
-            </ul></td>
-        </tr>
-    </tbody>
-</table>
-
-* Additional fields of a lesson aside from those stated in the table are used for recording cancelled dates of a lesson. More details can be found in [Editing a lesson](#editing-a-lesson--ledit).<br>
```
Comment (Xiaoyunnn) [inline docs/UserGuide.md:631]: why u delete this table tho?

File: docs/UserGuide.md [hunk 3]
```diff
@@ -542,121 +586,63 @@ A lesson can be categorised into 2 types:
...
+* End date
+* Cancelled dates
 
 TAB has features to help to prevent scheduling mistakes.
 For example, we help you avoid mistakes such as scheduling two lessons at the same slots by ensuring that lessons never overlap.
 Additionally, we help to avoid accidentally scheduling lessons outside working hours (before 0800 and after 2200 hours).
 
 In the future, we intend to add some nice-to-have features that will allow you to customise your working hours,
 as well as specify minimum/maximum lesson durations so that accidents like mistyping a 10-minute lesson won't happen.
-Additionally, we will allow you to toggle these checks on and off, just in case you don't want them. You would also be able to customise your own recurrence rule for your recurring lessons.
```
Comment (Xiaoyunnn) [inline docs/UserGuide.md:645]: why delete this?

File: docs/UserGuide.md [hunk 4]
```diff
@@ -43,7 +43,19 @@ _text_ | Caption for images.
...
+<div markdown="block" class="alert alert-info">
+
+**:information_source: JDK Installation Guide**
+
+You can install the required JDK and JRE from the
+[Java SE Development Kit Downloads page](https://www.oracle.com/java/technologies/downloads/).
+
+Here is a [website](https://docs.oracle.com/en/java/javase/17/install/overview-jdk-installation.html#GUID-8677A77F-231A-40F7-98B9-1FD0B48C346A)
+to help you with the installation. Follow the guide for your operating system for detailed instructions.
+</div>
```
Comment (eeliana) [inline docs/UserGuide.md:55]: I thought Amy said just have to give the 2nd website and no need to be so long?

File: docs/UserGuide.md [hunk 5]
```diff
@@ -414,6 +429,8 @@ Format: `delete INDEX`
 
 * Deletes the student at the specified `INDEX`.
 
+* The index must be a valid index number of a student shown in the displayed list.
```
Comment (eeliana) [inline docs/UserGuide.md:432]: This can be inferred from line 227, I think

File: docs/UserGuide.md [hunk 6]
```diff
@@ -440,79 +457,113 @@ This section guides you on how to find or filter students of your choice in TAB.
 #### Finding students by fields: `find`
 
 Finds all students whose fields match the given keyword(s), based on the specified find condition.
+You can find by tags, student fields and lesson fields.
```
Comment (eeliana) [inline docs/UserGuide.md:460]: Phrasing a bit strange here. Find what? Need some more time to think

File: docs/UserGuide.md [hunk 7]
```diff
@@ -440,79 +457,113 @@ This section guides you on how to find or filter students of your choice in TAB.
...
 
-* The search is case-insensitive.<br>
-  e.g. keyword `hans` will match `Hans`.
+##### Search by other student fields
+* You can specify one or more keywords for each parameter. e.g. `find n/Amad Ali`.
+* A field needs to contain all specified keywords to be matched.<br>
+  e.g. keywords `Amad Ali` will not match `Amad`, but it will match `Amad bin Ali`.
+* If you specify multiple of the same prefix, only the last prefix will be used for the search of that field.
+* A keyword can match a word partially. e.g. keyword `uni` will match `university`.
+* The search is case-insensitive.
```
Comment (eeliana) [inline docs/UserGuide.md:482]: Super minor thing but line 481,482 not in same order as line 470,471 ahaha

File: docs/UserGuide.md [hunk 8]
```diff
@@ -440,79 +457,113 @@ This section guides you on how to find or filter students of your choice in TAB.
...
+<table id="find-param-table">
+    <thead>
+        <tr>
+            <th style="text-align:center">Fields</th>
+            <th style="text-align:center">Find behaviour</th>
+        </tr>
+    </thead>
+    <tbody>
+        <tr>
+            <td>Subject <br> Lesson Rate</td>
```
Comment (eeliana) [inline docs/UserGuide.md:499]: I thought we standardised to "Lesson rates" to match the prefixes?

File: docs/UserGuide.md [hunk 9]
```diff
@@ -440,79 +457,113 @@ This section guides you on how to find or filter students of your choice in TAB.
...
+              <ul>
+                <li>The same constraints for student fields apply.</li>
+              </ul>
+            </td>
+        </tr>
+        <tr>
+            <td>Homework</td>
+            <td>
+              <ul>
+                <li>Finds students with lessons that have at least one homework that matches all the keywords.</li>
```
Comment (eeliana) [inline docs/UserGuide.md:510]: Not sure about this phrasing/suggestion but the phrasing here seems a bit awkward

```suggestion
                <li>Finds students with lessons that have at least one piece of homework matching all the keywords.</li>
```

File: docs/UserGuide.md [hunk 10]
```diff
@@ -440,79 +457,113 @@ This section guides you on how to find or filter students of your choice in TAB.
...
+                <li>The same constraints for student fields apply.</li>
+              </ul>
+            </td>
+        </tr>
+        <tr>
+            <td>Start Date</td>
+            <td>
+              <ul>
+                <li>Accepts one keyword that must follow the Date format</li>
```
Comment (eeliana) [inline docs/UserGuide.md:519]: Link to what Date format we talking about?
As specified in [Managing Lessons](#managing-lessons) ?

File: docs/UserGuide.md [hunk 11]
```diff
@@ -440,79 +457,113 @@ This section guides you on how to find or filter students of your choice in TAB.
...
+                <li>Finds students with lessons that are cancelled on the specified date.</li>
+              </ul>
+            </td>
+        </tr>
+        <tr>
+            <td>Time Range</td>
+            <td>
+              <ul>
+                <li>Accepts one keyword that must follow the Time Range format</li>
+                <li>Finds students with lessons that occur within the time range, excluding the start and end time.<br>
```
Comment (eeliana) [inline docs/UserGuide.md:538]: Is it within or overlapping? I'm not too sure about this

File: docs/UserGuide.md [hunk 12]
```diff
@@ -542,121 +586,63 @@ A lesson can be categorised into 2 types:
...
-</div>
-
-<table id="lesson-param-table">
-    <thead>
-        <tr>
-            <th style="text-align:center; padding: 10px">Category</th>
-            <th style="text-align:center">Field</th>
-            <th style="text-align:center">Prefix</th>
-            <th style="text-align:center">Constraints</th>
-            <th style="text-align:center">Examples</th>
```
Comment (eeliana) [inline docs/UserGuide.md:558]: I'm a little confused.. you want to remove the table?

File: docs/UserGuide.md (hunk 6 above)
Comment (Xiaoyunnn) [inline docs/UserGuide.md:460]:
```suggestion
You can find students by tags, student fields and lesson fields.
```
not sure if this sounds clearer

**Comment by lingshanng** [comment]
oops sorry, seems like got some issue when I merged

File: docs/UserGuide.md (hunk 7 above)
Comment (lingshanng) [reply docs/UserGuide.md:472]: what do you mean by this?

Comment (eeliana) [reply docs/UserGuide.md:472]: In 470,471, the case sensitivity point comes after how the word is matched

File: docs/UserGuide.md [hunk 13]
```diff
@@ -436,11 +423,9 @@ Format: `delete INDEX`
 
 * Deletes the student at the specified `INDEX`.
 
-* The index must be a valid index number of a student shown in the displayed list.
-
 Examples:
 * `list` followed by `delete 2` deletes the 2nd student in TAB.
-* `find n/Betsy` followed by `delete 1` deletes the 1st student in the results of the `find` command.
+* `find n/Betsy` followed by `ldelete 1 1` deletes the 1st lesson for the 1st student in the results of the `find` command.
```
Comment (eeliana) [inline docs/UserGuide.md:428]: This is `delete` not `ldelete` right?

Comment (Xiaoyunnn) [reply docs/UserGuide.md:428]: i tot this is saying `find n/Betsy` filters the list of students then `ldelete` acts on the 1st student 's 1st lesson in filtered list

Comment (eeliana) [reply docs/UserGuide.md:428]: But this example is under the "Deleting a student" section though, maybe should go under "Find" section instead?

Comment (Xiaoyunnn) [reply docs/UserGuide.md:428]: oh right, j saw a duplicate eg under `ldelete` section

**Review by Xiaoyunnn — APPROVED** [review]
LGTM

**Review by eeliana — CHANGES_REQUESTED** [review]
Can you add the "Find" section to the TOC in "Managing Students"?

**Review by eeliana — APPROVED** [review]

**Pull request merged**